src/main/java/org/reactome/release/cosmicupdate/Main.java

```diff
@@ -0,0 +1,337 @@
+	 * an IllegalArgumentException is thrown.
+	 * @param value The value to check
+	 * @param message The message that will be put into the IllegalArgumentException
+	 * @throws IllegalArgumentException If value == null || value.isEmpty()
+	 */
+	private void validateConfigValue(String value, String message)
+	{
+		if (value == null || value.trim().isEmpty())
+		{
+			throw new IllegalArgumentException(message);
+		}
+	}
+	
+	/**
+	 * Download the data files from COSMIC.
+	 * @throws Exception
+	 */
+	private void downloadFiles() throws Exception
+	{
+		//TODO: Better Properties class in release-common-lib. (for future work)
+		validateConfigValue(Main.COSMICUsername, "COSMIC Username cannot be null/empty! Please set a value for cosmic.username in the application's properties file");
+		validateConfigValue(Main.COSMICUsername, "COSMIC Password cannot be null/empty! Please set a value for cosmic.password in the application's properties file");
+		validateConfigValue(Main.COSMICMutantExportURL, "URL for COSMIC Mutant Export file cannot be null/empty! Please set a value for urlToMutantExportFile in the application's properties file");
+		validateConfigValue(Main.COSMICMutationTrackingURL, "URL for COSMIC Mutation Tracking file cannot be null/empty! Please set a value for urlToMutationTrackingFile in the application's properties file");
+		validateConfigValue(Main.COSMICFusionExportURL, "URL for COSMIC Fusion Export file cannot be null/empty! Please set a value for urlToFusionExportFile in the application's properties file");
+		
+		COSMICFileRetriever mutantExportRetriever = new COSMICFileRetriever();
+		COSMICFileRetriever mutationTrackingRetriever = new COSMICFileRetriever();
+		COSMICFileRetriever fusionExportRetriever = new COSMICFileRetriever();
+		
+		try
+		{
+			mutantExportRetriever.setDataURL(new URI(Main.COSMICMutantExportURL));
+			mutationTrackingRetriever.setDataURL(new URI(Main.COSMICMutationTrackingURL));
+			fusionExportRetriever.setDataURL(new URI(Main.COSMICFusionExportURL));
+			
+			final String mutantExportDestination = "./CosmicMutantExport.tsv.gz";
+			final String mutationTrackingDestination = "./CosmicMutationTracking.tsv.gz";
+			final String fusionExportDestination = "./CosmicFusionExport.tsv.gz";
+			
+			mutantExportRetriever.setFetchDestination(mutantExportDestination);
+			mutationTrackingRetriever.setFetchDestination(mutationTrackingDestination);
+			fusionExportRetriever.setFetchDestination(fusionExportDestination);
+			
+			logDownloadMessage(mutantExportRetriever, mutantExportDestination);
+			this.executeDownload(mutantExportRetriever);
+			
+			logDownloadMessage(mutationTrackingRetriever, mutationTrackingDestination);
+			this.executeDownload(mutationTrackingRetriever);
+			
+			logDownloadMessage(fusionExportRetriever, fusionExportDestination);
+			this.executeDownload(fusionExportRetriever);
+		}
+		catch (URISyntaxException e)
+		{
+			e.printStackTrace();
+			throw e;
```

Comment:
Is a stacktrace needed here if the exception is rethrown and dealt with later?

Reply:
Exception handling is being re-worked...